tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py

```diff
@@ -47,11 +47,19 @@ def __init__(self, **kwargs):
         self._temp_file_name = "tab_trans_temp.pth"
         self._period_columns_mapping = None
 
-        # TODO: Take in num_gpu's as a param. Currently this is hard-coded upon detection of cuda.
-        if torch.cuda.is_available():
-            self.device = torch.device("cuda")
-        else:
+        num_gpus = self.params.get('num_gpus', None)
```

Comment:
`num_gpus` is not in `params`, it is a kwarg of `_fit()`. See here for example:

https://github.com/awslabs/autogluon/blob/master/tabular/src/autogluon/tabular/models/tabular_nn/tabular_nn_model.py#L194

Reply:
Moved to _fit and changed self.params to kwargs.